subprojects/reporting/src/main/java/org/gradle/api/reporting/Report.java

```diff
@@ -95,6 +95,8 @@ public String determineName(Report report) {
     @Internal
     File getDestination();
 
+    void setDestination(Object file);
```

Comment:
this is already on `ConfigurableReport`. There should also be an overload that takes `Provider<File>` and `File` and we should deprecate `setDestination(Object)`